jdbc/sqlserver/src/test/java/com/google/cloud/sql/sqlserver/JdbcSqlServerUnitTests.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.sql.sqlserver;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.sql.core.CoreSocketFactory;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class JdbcSqlServerUnitTests {
+
+  private static final String CONNECTION_NAME = "my-project:my-region:my-instance";
+
+  @Test
+  public void checkConnectionStringNoQueryParams() {
+    String socketFactoryConstructorArg = CONNECTION_NAME;
+    SocketFactory socketFactory = new SocketFactory(socketFactoryConstructorArg);
+    assertThat(socketFactory.props.get(CoreSocketFactory.CLOUD_SQL_INSTANCE_PROPERTY)).isEqualTo(
+        CONNECTION_NAME);
+  }
+
+  @Test
+  public void checkConnectionStringWithQueryParam() {
+    String socketFactoryConstructorArg = String.format("%s?%s=%s", CONNECTION_NAME, "ipTypes", "PRIVATE");
+    SocketFactory socketFactory = new SocketFactory(socketFactoryConstructorArg);
+    assertThat(socketFactory.props.get(CoreSocketFactory.CLOUD_SQL_INSTANCE_PROPERTY)).isEqualTo(
+        CONNECTION_NAME);
+    assertThat(socketFactory.props.get("ipTypes")).isEqualTo(
+        "PRIVATE");
+  }
+
```

Comment:
I would love to see test that try out some malicious input. 

Here are a few to get you started:
```
"proj:inst:db?"
   expected: props.size() == 0, no NPE, no IndexOutOfBoundsExeptions

"proj:inst:db?useIamAuth=true&privateIp" 
    expected: useIamAuth=="true", privateIp==""

"proj:inst:db?token=abc%20def%20xyz%2F%26%3D%20%48%49%20%53%68%75%62%61"  
    expected: token == "abc def xyz/&= HI Shuba"
```